fluidsynth/src/synth/fluid_chan.h

```diff
@@ -41,7 +41,7 @@ struct _fluid_channel_t
   int sfont_bank_prog;                  /**< SoundFont ID (bit 21-31), bank (bit 7-20), program (bit 0-6) */
   fluid_preset_t* preset;               /**< Selected preset */
 
-  int key_pressure;                     /**< MIDI key pressure */
+  int key_pressure[128];                /**< MIDI polyphonic key pressure */
```

Comment:
Do we really need `int` here? Makes a worst case memory usage of `8*128 bytes = 1KiB`. I think internally we can use `signed char`.

Note however that the types you've chosen for the API calls can stay as they are, i.e. `fluid_event_key_pressure` with `short` and `fluid_synth_key_pressure` with `int`.